File: apps/football_buddy/lib/football_buddy/application/football_core/competition.ex
defmodule FootballBuddy.FootballCore.Competition do
  defstruct [:name,
             :code,
             :year,
             :matchdays_count,
             :teams_count]

  def build(data) do
    %__MODULE__{
      name: data["name"],
      code: data["code"],
      year: data["year"],
      matchdays_count: data["matchdays_count"],
      teams_count: data["teams_count"]
    }
  end
end

File: apps/football_buddy_web/frontend/src/data/domain/Competition.js
import { Record } from 'immutable'

const competitionObject = {
  code: null,
  name: null,
  year: null,
  matchdaysCount: null,
  teamsCount: false,
}

const CompetitionRecord = Record(competitionObject)

class Competition extends CompetitionRecord {
  // Extend immutable js Record
}

export function parseCompetition(object) {
  return new Competition({
    code: object.code,
    name: object.name,
    year: object.yeat,
    matchdaysCount: object.matchdays_Count,
    teamsCount: object.teams_count,
  })
}

export default Competition
